src/AWS.Deploy.CLI/Commands/ServerModeCommand.cs

```diff
@@ -16,26 +19,75 @@ public class ServerModeCommand
         private readonly IToolInteractiveService _interactiveService;
         private readonly int _port;
         private readonly int? _parentPid;
+        private readonly bool _encryptionKeyInfoStdIn;
 
-        public ServerModeCommand(IToolInteractiveService interactiveService, int port, int? parentPid)
+        public ServerModeCommand(IToolInteractiveService interactiveService, int port, int? parentPid, bool encryptionKeyInfoStdIn)
         {
             _interactiveService = interactiveService;
             _port = port;
             _parentPid = parentPid;
+            _encryptionKeyInfoStdIn = encryptionKeyInfoStdIn;
         }
 
         public async Task ExecuteAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
+            IEncryptionProvider encryptionProvider = CreateEncryptionProvider();
+
             var url = $"http://localhost:{_port}";
 
             var builder = new WebHostBuilder()
                 .UseKestrel()
                 .UseUrls(url)
+                .ConfigureServices(services =>
+                {
+                    services.AddSingleton<IEncryptionProvider>(encryptionProvider);
+                })
                 .UseStartup<Startup>();
 
             var host = builder.Build();
 
             await host.RunAsync(cancellationToken); 
+         }
+
+        private IEncryptionProvider CreateEncryptionProvider()
+        {
+            IEncryptionProvider encryptionProvider;
+            if (_encryptionKeyInfoStdIn)
+            {
+                _interactiveService.WriteLine("Waiting on encryption key info from stdin");
+                var input = _interactiveService.ReadLine();
+                var keyInfo = EncryptionKeyInfo.ParseStdInKeyInfo(input);
```

Comment:
Should we make `EncryptionKeyInfo` more like ComplexInput type where we can ask customer to individual fields then asking for a complex input?

Reply:
No, the customer is the IDE and they are not reading stdout to know what properties to set. Having a single input blob makes it easier for the IDE to know it was when stdin write to do.